From 685425ebb596067265734f143565b31917cad117 Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Sat, 11 Jan 2020 03:16:03 +0100
Subject: [PATCH] cmake: Fix Waffle detection

The commit 36e1ae064ffc ("cmake: remove FindWaffle.cmake") in Waffle
removed FindWaffle.cmake back in 2014 . Apply the same Waffle detection
mechanism as Piglit instead .

Signed-off-by: Marek Vasut <marex@denx.de>
---
 CMakeLists.txt | 23 ++++++++++++++++++++++-
 1 file changed, 22 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 19316e60..f491693b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -192,8 +192,29 @@ elseif (ENABLE_X11)
 endif ()
 
 if (ENABLE_EGL AND ENABLE_WAFFLE)
+    include (FindPkgConfig)
     # Use Waffle for eglretrace
-    find_package (Waffle REQUIRED)
+    if (NOT WIN32)
+        pkg_check_modules(Waffle REQUIRED waffle-1)
+
+        if(NOT Waffle_FOUND)
+            message(FATAL_ERROR "Failed to find Waffle. If Waffle "
+            "is not packaged for your distribution, you can get "
+            "it at http://www.waffle-gl.org."
+        )
+        endif()
+    else ()
+        find_path(Waffle_INCLUDE_DIRS waffle.h)
+        find_library(Waffle_LDFLAGS waffle-1)
+        if(Waffle_INCLUDE_DIRS AND Waffle_LDFLAGS)
+            set(Waffle_FOUND TRUE)
+        else()
+            message(FATAL_ERROR "Failed to find Waffle. Get and build Waffle from "
+                "http://www.waffle-gl.org and set Waffle_INCLUDE_DIRS and "
+                "Waffle_LDFLAGS variables accordingly."
+            )
+        endif()
+    endif ()
 endif ()
 
 if (ENABLE_EGL AND NOT ENABLE_X11)
-- 
2.24.1

